### PR TITLE
Fix two issues with ResetBlockFailureFlags, add tests

### PR DESCRIPTION
**Pull request by UdjinM6**
Without fe38ff9 new test crashes with errors like
```
Assertion failure:
  assertion: (pindex->nStatus & BLOCK_FAILED_MASK) == 0
  file: validation.cpp, line: 5013
  function: CheckBlockIndex
   0#: (0x10E526E65) stacktraces.cpp:651        - __assert_rtn
   1#: (0x10E2BBBD0) validation.cpp:5013        - CheckBlockIndex
   2#: (0x10E2BA6DC) microsec_time_clock.hpp:70 - ActivateBestChain
   3#: (0x10E2BBF34) validation.cpp             - ???
   4#: (0x10E181B2A) memory:4521                - reconsiderblock
   5#: (0x10E21B72B) server.cpp:626             - execute
   6#: (0x10DE65A0D) httprpc.cpp:197            - HTTPReq_JSONRPC
   7#: (0x10DE73178) functional:1799            - operator()
   8#: (0x10DE746BD) memory:2651                - Run
   9#: (0x10DE74C8A) memory:2649                - __thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (*)(WorkQueue<HTTPClosure> *), WorkQueue<HTTPClosure> *> >
  10#: (0x7FFF6DDC82EB) <unknown-file>             - ???
  11#: (0x7FFF6DDCB249) <unknown-file>             - ???
  12#: (0x7FFF6DDC740D) <unknown-file>             - ???
Assertion failed: ((pindex->nStatus & BLOCK_FAILED_MASK) == 0), function CheckBlockIndex, file validation.cpp, line 5013.
```

### CODE REVIEW

File: test/functional/rpc_invalidateblock.py [hunk 1]
```diff
@@ -60,5 +60,31 @@ def run_test(self):
...
+            self.restart_node(0, extra_args=["-checkblocks=5"])
+            tip = self.nodes[0].generate(10)[-1]
+            self.nodes[1].generate(9)
+            connect_nodes(self.nodes[0], 1)
+            self.sync_blocks(self.nodes[0:2])
+            assert_equal(self.nodes[0].getblockcount(), newheight + 10 * (j + 1))
+            assert_equal(self.nodes[1].getblockcount(), newheight + 10 * (j + 1))
+            assert_equal(self.nodes[1].getbestblockhash(), tip)
+
+        tip = self.nodes[1].getbestblockhash()
```
Comment (xdustinface) [inline test/functional/rpc_invalidateblock.py:80]:
```suggestion
```

Seems to be a leftover?

Comment (UdjinM6) [reply test/functional/rpc_invalidateblock.py:80]: I actually forgot to include a test for it 🙈 see 074079225b

**Review by xdustinface — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK

**Pull request merged**